### ISSUE TITLE
another bug over ZZ

### ISSUE THREAD

**Issue by jakobkroeker** [Macaulay2/M2#315]
finally I found some time to test Macaulays computation over integers a little bit.
Here is the next bug:

```
R = ZZ[x, y, MonomialOrder => Lex]
I = ideal(32,16*x^2*y-10*x*y+1*x)
  gI = ideal gens gb I;
  assert( ( (gens I)%gI) == 0 );
```

output

```
i1 : R = ZZ[x, y, MonomialOrder => Lex]

o1 = R

o1 : PolynomialRing

i2 : I = ideal(32,16*x^2*y-10*x*y+1*x)

                   2
o2 = ideal (32, 16x y - 10x*y + x)

o2 : Ideal of R

i3 :   gI = ideal gens gb I;

o3 : Ideal of R

i4 :   assert( ( (gens I)%gI) == 0 );
stdio:4:3:(3): error: assertion failed

i5 : gI

o5 = ideal (32, 2x)

o5 : Ideal of R


```

 `gI= 32,2*x` seems wrong (and Singular tells `gI= 32,x`)

**Comment by jakobkroeker** [comment]
more complicated failing example:
```
R = ZZ[x, y, MonomialOrder => Lex]
I = ideal(2*x^2*y^2,20*x*y^3-10*x*y^2-1,x^2-22*x*y)
  gI = ideal gens gb I;
  assert( ( (gens I)%gI) == 0 );
(gens I)%gI
gI
```
output
```
i4 :   assert( ( (gens I)%gI) == 0 );
stdio:4:3:(3): error: assertion failed

i5 : (gens I)%gI

o5 = | 0 1 0 |

             1       3
o5 : Matrix R  <--- R

i6 : gI

o6 = ideal (2, y + 1, x)

o6 : Ideal of R

```

Singular tells me that  `I = 1`

**Comment by jakobkroeker** [comment]
las one: 
```
R = ZZ[x, y, MonomialOrder => Lex]
I = ideal(22*y^3,-22*x^2*y-13*x-2*y,-26*x*y^2-x)
  gI = ideal gens gb I;
  assert( ( (gens I)%gI) == 0 );
(gens I)%gI
gI

```
output
```
i4 :   assert( ( (gens I)%gI) == 0 );
stdio:4:3:(3): error: assertion failed

i5 : (gens I)%gI

o5 = | 0 -8y3-2y 0 |

             1       3
o5 : Matrix R  <--- R

i6 : gI

               2    4     2        3
o6 = ideal (22y , 2y  + 6y , x - 4y )

o6 : Ideal of R
```
Singular tells
```
gI[1]=22y
gI[2]=2y3+6y
gI[3]=x-10y

```

**Comment by mikestillman** [comment]
Fixed in commit f701d87b89faf75.  This was a serious bug, sometimes giving incorrect answers.  The fix, though simple, might have efficiency repercussions, although there are larger examples that now run (correctly and) much faster.
